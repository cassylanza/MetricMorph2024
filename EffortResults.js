import React, { Component } from 'react'
import { TouchableOpacity, Text, View, StyleSheet, ScrollView } from 'react-native'
import { Actions } from 'react-native-router-flux'


// formats the calculated time into hours, minutes, and seconds
// returns default of "hh:mm:ss" if input is invalid (i.e. not a number
// or outside of the specified range)
export function formatResult(timeInSeconds) {

  let maxNumber = 362399;

  if (isNaN(timeInSeconds) || typeof timeInSeconds !== 'number') {
      return "hh:mm:ss";
  }

  if(timeInSeconds < 0 || timeInSeconds > maxNumber) {
    return "hh:mm:ss";
  }

  let hours = parseInt((Number(timeInSeconds))/3600);
  let minutes = parseInt(Number(timeInSeconds)/60)-(hours*60);
  let seconds = (Number(timeInSeconds)%60);
  let formattedSplit = timeInSeconds;


  if (timeInSeconds < 60) {
    formattedSplit = seconds.toFixed(2);
  }
  else if (timeInSeconds < 60*60) {

    if (seconds < 10) {
      formattedSplit = minutes + ":0" + seconds.toFixed(2);
    } else {
      formattedSplit = minutes + ":" + seconds.toFixed(2);
    }
  }
  else {

    formattedSplit = hours;

    if (minutes < 10) {
      formattedSplit = formattedSplit + ":0" + minutes;
    } else {
      formattedSplit = formattedSplit + ":" + minutes;
    }

    if (seconds < 10) {
      formattedSplit = formattedSplit + ":0" + seconds.toFixed(2);
    } else {
      formattedSplit = formattedSplit + ":" + seconds.toFixed(2);
    }
  }
  return formattedSplit;
}


// main display component
// calculates a pace corresponding to an effort percentage
// converts given performance to 3200 performance and divides by given % effort
// returns main calculation with table underneath
const EffortResults = ({race, effortPercent, hours, minutes, seconds}) => {

  let basisDistance = 3200; // effort relies on 3200 pace

  let conversionExponent = 1.06; // value of exponent in Riegal's conversion formula

  let currentRaceTimeVar = Number(hours)*3600 + Number(minutes)*60 + Number(seconds);

  let mySplit = Number(currentRaceTimeVar) * (Number(basisDistance)/(Number(race))) ** (conversionExponent);

  const events = [
      {
        id: 0,
        effort: '65%',
        time: formatResult(Number(mySplit/2)/(Number(.65))),
      },
      {
        id: 1,
        effort: '70%',
        time: formatResult(Number(mySplit/2)/(Number(.70))),
      },
      {
        id: 2,
        effort: '75%',
        time: formatResult(Number(mySplit/2)/(Number(.75))),
      },
      {
        id: 3,
        effort: '80%',
        time: formatResult(Number(mySplit/2)/(Number(.8)))
      },
      {
        id: 4,
        effort: '85%',
        time: formatResult(Number(mySplit/2)/(Number(.85)))
      },
      {
        id: 5,
        effort: '90%',
        time: formatResult(Number(mySplit/2)/(Number(.9)))
      },
      {
        id: 6,
        effort: '95%',
        time: formatResult(Number(mySplit/2)/(Number(.95)))
      },
      {
        id: 7,
        effort: '96%',
        time: formatResult(Number(mySplit/2)/(Number(.96)))
      },
      {
        id: 8,
        effort: '97%',
        time: formatResult(Number(mySplit/2)/(Number(.97)))
      },
      {
        id: 9,
        effort: '98%',
        time: formatResult(Number(mySplit/2)/(Number(.98)))
      },
      {
        id: 10,
        effort: '99%',
        time: formatResult(Number(mySplit/2)/(Number(.99)))
      },
      {
        id: 11,
        effort: '100%',
        time: formatResult(Number(mySplit/2)/(Number(1)))
      },
      {
        id: 12,
        effort: '105%',
        time: formatResult(Number(mySplit/2)/(Number(1.05)))
      },
      {
        id: 13,
        effort: '110%',
        time: formatResult(Number(mySplit/2)/(Number(1.1)))
      },
      {
        id: 14,
        effort: '115%',
        time: formatResult(Number(mySplit/2)/(Number(1.15)))
      },
      {
        id: 15,
        effort: '120%',
        time: formatResult(Number(mySplit/2)/(Number(1.2)))
      }
    ];

    return (
      // displays main converion result followed by a table of
      // common effort percentages
      <View>
        <Text style = {styles.mainResult}> { formatResult((mySplit/(Number(2)))/(Number(effortPercent)/(Number(100)))) } </Text>
        <Text style = {{textAlign: 'center', marginBottom: 30}}> min/mile for { effortPercent }% </Text>
        <ScrollView>
          {events.map((item, index) => (
              <View key={item.id} style={styles.container}>
                <Text style={styles.text}>{item.effort}</Text>
                <Text style={styles.text}>{item.time}</Text>
              </View>
            ))}
        </ScrollView>
      </View>
    )
}
export default EffortResults

const styles = StyleSheet.create ({
  container: {
      padding: 10,
      marginTop: 3,
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'flex-end',
      borderWidth: 1
   },
   text: {
      color: '#4f603c'
   },
   mainResult: {
     width: 250,
     height: 75,
     textAlign: 'center',
     padding: 18,
     fontSize: 30,
     backgroundColor: 'white',
     marginTop: 60,
     marginBottom: 10,
     marginLeft: 60
   }
})
