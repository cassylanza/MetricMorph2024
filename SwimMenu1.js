import React from 'react'
import { TouchableOpacity, Text, View, StyleSheet, Image } from 'react-native'
import { Actions } from 'react-native-router-flux'

const SwimMenu1 = () => {

  const goToTrackMenu = () => {
    Actions.trackmenu1()
  }

  const goToSwimMenu = () => {
    Actions.swimmenu1()
  };

  const goToConvertMeterYardEntry = () => {
    Actions.convmeteryardinfoentry()
  }

  const goToHundredProjectionInfoEntry = () => {
    Actions.hundredinfoentry()
  }

  const goToRaceProjectionInfoEntry = () => {
    Actions.raceinfoentry()
  }

  return (
    <View style = {styles.containerScreen}>
      <Image style = {styles.logo} source={require('./MetricMorphLogo.png')} />
      <View style = {styles.swimMenuButton}>
        <TouchableOpacity onPress = {goToConvertMeterYardEntry}>

         <Text style = {styles.textSwimMenu}>Meter/yard conversion</Text>

        </TouchableOpacity>
</View>
<View style = {styles.swimMenuButton}>
        <TouchableOpacity onPress = {goToHundredProjectionInfoEntry}>
         <Text style = {styles.textSwimMenu}>6x100 Projection</Text>
        </TouchableOpacity>
</View>
<View style = {styles.swimMenuButton}>
        <TouchableOpacity onPress = {goToRaceProjectionInfoEntry}>

         <Text style = {styles.textSwimMenu}>Race Projection</Text>

        </TouchableOpacity>
        </View>

       <View style = {styles.containerTaskBar}>
          <TouchableOpacity onPress = {goToSwimMenu}>
            <Text style = {styles.textTaskBarSwim}> Swim Menu </Text>
          </TouchableOpacity>
          <TouchableOpacity onPress = {goToTrackMenu}>
            <Text style = {styles.textTaskBar}> Track Menu </Text>
          </TouchableOpacity>
        </View>
      </View>

  )
}
export default SwimMenu1

const styles = StyleSheet.create ({
   containerScreen: {
     flexDirection: 'column',
     alignItems: 'center',
     backgroundColor: 'white',
    flex: 1
   },

   logo: {
      marginTop: 80,
      marginBottom: 75
   },

   swimMenuButton: {
  padding: 10,
  margin: 20,
  width: 280,
  height: 60,
  borderRadius: 25,
  alignItems: 'center',
  justifyContent: 'center',
  flexDirection: 'center',
  backgroundColor: 'green'

   },

  containerTaskBar: {
     flex: 1,
     flexDirection: 'row',
     position: 'absolute',
     alignItems: 'center',
     height: '12%',
     width: '100%',
     bottom: 0,
     left: 0,
     borderWidth: 1,
     backgroundColor: 'white'
  },

   textTaskBar: {
     width: 188,
     height: 100,
     textAlign: 'center',
     padding: 37,
     fontSize: 20,


   },
   textTaskBarSwim: {
     width: 188,
     height: 100,
     textAlign: 'center',
     padding: 37,
     fontSize: 20,
     fontWeight: 'bold',
     color: 'blue'
   },
   textSwimMenu: {
      textAlign: 'center',
      fontSize: 20,
      color: 'white'
   }
})
