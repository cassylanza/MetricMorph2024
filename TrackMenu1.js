import React from 'react'
import { TouchableOpacity, Text, View, StyleSheet, Image } from 'react-native'
import { Actions } from 'react-native-router-flux'
import { FontAwesomeIcon }  from '@fortawesome/react-native-fontawesome'
import { faPersonSwimming } from '@fortawesome/free-solid-svg-icons/faPersonSwimming'
import { faPersonRunning } from '@fortawesome/free-solid-svg-icons/faPersonRunning'


const TrackMenu1 = () => {

  // navigation constants
  const goToSwimMenu = () => {
    Actions.swimmenu1()
  }

  const goToConvertInfoEntry = () => {
    Actions.convperfinfoentry()
  }

  const goToSplitsInfoEntry = () => {
    Actions.splitsinfoentry()
  }

  const goToEffortInfoEntry = () => {
    Actions.effortinfoentry()
  }

  return (
    <View style = {styles.containerScreen}>
    {/*  <Image style = {styles.logo} source={require('./MetricMorphLogo.png')} /> */}
      <View style = {{padding: '50%'}}>
        <TouchableOpacity onPress = {goToConvertInfoEntry} style = {styles.trackMenuButton}>

           <Text style = {styles.textTrackMenu}>Convert Race</Text>

        </TouchableOpacity>

        <TouchableOpacity onPress = {goToSplitsInfoEntry} style = {styles.trackMenuButton}>

           <Text style = {styles.textTrackMenu}>Splits</Text>

          </TouchableOpacity>

        <TouchableOpacity onPress = {goToEffortInfoEntry} style = {styles.trackMenuButton}>

           <Text style = {styles.textTrackMenu}>% Effort</Text>

          </TouchableOpacity>
      </View>

      <View style = {styles.containerTaskBar}>
        <TouchableOpacity onPress = {goToSwimMenu} style={{alignItems: 'center', marginBottom: '2%'}}>
            <FontAwesomeIcon
              icon={faPersonSwimming}
              size={30}
              color='gray'
              style = {{marginBottom: 2}}
            />
            <Text style = {{color: 'gray'}}>Swimming</Text>
        </TouchableOpacity>
        <View style = {{alignItems: 'center', marginBottom: '2%'}}>
          <FontAwesomeIcon
            icon={faPersonRunning}
            size={30}
            color='#00b3b3'
            style = {{marginBottom: 2}}
          />
          <Text style = {{color: '#00b3b3'}}>Track</Text>
        </View>
      </View>
    </View>
  )
}
export default TrackMenu1

const styles = StyleSheet.create ({
   containerScreen: {
     flexDirection: 'column',
     alignItems: 'center',
     backgroundColor: 'white',
     flex: 1
   },
   logo: {
      marginTop: 80,
      marginBottom: 75
   },
   trackMenuButton: {
     padding: 10,
     margin: 20,
     width: 280,
     height: 60,
     borderRadius: 25,
     alignItems: 'center',
     justifyContent: 'center',
     flexDirection: 'center',
     backgroundColor: '#00b3b3'
   },
   textTrackMenu: {
      textAlign: 'center',
      fontSize: 20,
      color: 'white'
   },
   containerTrackMenu: {
     textAlign: 'center',
     flexDirection: 'column',
     justifyContent: 'center',
     alignItems: 'center',
     backgroundColor: 'white',
     height: 420,
     width: 300
   },
   containerTaskBar: {
     flex: 1,
     flexDirection: 'row',
     position: 'absolute',
     justifyContent: 'space-around',
     alignItems: 'center',
     height: '12%',
     width: '100%',
     bottom: 0,
     left: 0,
     borderWidth: 1,
     borderTopColor: 'lightgray',
     backgroundColor: 'white'
   },
   textTaskBar: {
     width: 188,
     height: 100,
     textAlign: 'center',
     padding: 37,
     fontSize: 20,
   }
})
