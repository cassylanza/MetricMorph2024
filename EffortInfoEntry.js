import React, { Component, useState } from 'react'
import { TouchableOpacity, Text, View, StyleSheet, TextInput, Alert, ScrollView } from 'react-native'
import { Actions } from 'react-native-router-flux'
import EffortResults from './EffortResults.js'
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome'
import { faPersonSwimming } from '@fortawesome/free-solid-svg-icons/faPersonSwimming'
import { faPersonRunning } from '@fortawesome/free-solid-svg-icons/faPersonRunning'

const EffortInfoEntry = () => {

  // states for input fields
  const [hours, setHours] = useState('');
  const [minutes, setMinutes] = useState('');
  const [seconds, setSeconds] = useState('');
  const [race, setRace] = useState('');
  const [effortPercent, setEffortPercent] = useState('');
  const [showResults, setShowResults] = useState(false);

  // navigation constants
  const goToSwimMenu = () => {
    Actions.swimmenu1()
  }
  const goToTrackMenu = () => {
    Actions.trackmenu1()
  }
  const calculateEffort = () => {
      setShowResults(true);
  };

return (
  // display info entry fields and task bar
  <View>
    <ScrollView>
      <View style = {styles.container}>
        <TextInput style = {styles.input}
          placeholder = " Current Race Distance (in meters)"
          placeholderTextColor = "grey"
          keyboardType='number-pad'
          onChangeText = {newRace => setRace(newRace)}
          maxLength = {15}/>

          <View style = {styles.timeInputContainer}>
            <TextInput style = {styles.input}
              placeholder = "  hh  "
              keyboardType = 'number-pad'
              placeholderTextColor = "grey"
              onChangeText = {newHours => setHours(newHours)}
              maxLength = {2}/>

            <Text> : </Text>

            <TextInput style = {styles.input}
              placeholder = "  mm  "
              keyboardType = 'number-pad'
              placeholderTextColor = "grey"
              onChangeText = {newMinutes => setMinutes(newMinutes)}
              maxLength = {2}/>

            <Text> : </Text>

            <TextInput style = {styles.input}
              placeholder = "  ss  "
              keyboardType = 'decimal-pad'
              placeholderTextColor = "grey"
              onChangeText = {newSeconds => setSeconds(newSeconds)}
              maxLength = {5}/>

          </View>


        <TextInput style = {styles.input}
          placeholder = " Effort percentage (1-120%)"
          placeholderTextColor = "grey"
          keyboardType='number-pad'
          onChangeText = {newEffortPercent => setEffortPercent(newEffortPercent)}
          maxLength = {15}/>

        <TouchableOpacity
          style = {styles.convertButton}
          onPress = {calculateEffort}
          >
          <Text style = {styles.convertButtonText}> Calculate </Text>
        </TouchableOpacity>

        {/* reveal result(s) and table upon pressing convert button*/}
        {showResults && (
          <View>
            <EffortResults
              race={race}
              hours={hours}
              minutes={minutes}
              seconds={seconds}
              effortPercent={effortPercent}
            />
          </View>
        )}

      </View>
    </ScrollView>

    <View style = {styles.containerTaskBar}>
    <TouchableOpacity onPress = {goToSwimMenu} style={{alignItems: 'center', marginBottom: '2%'}}>
        <FontAwesomeIcon
          icon={faPersonSwimming}
          size={30}
          color='gray'
          style = {{marginBottom: 2}}
        />
        <Text style = {{color: 'gray'}}>Swimming</Text>
    </TouchableOpacity>
    <TouchableOpacity onPress = {goToTrackMenu} style = {{alignItems: 'center', marginBottom: '2%'}}>
      <FontAwesomeIcon
        icon={faPersonRunning}
        size={30}
        color='#00b3b3'
        style = {{marginBottom: 2}}
      />
      <Text style = {{color: '#00b3b3'}}>Track</Text>
    </TouchableOpacity>
    </View>

  </View>
)
}
export default EffortInfoEntry

const styles = StyleSheet.create({
container: {
  paddingTop: 23,
  marginBottom: '100%'
},
input: {
  margin: '10%',
  height: 50,
  borderBottomWidth: 1,
  borderBottomColor: 'grey'
},
convertButton: {
  backgroundColor: '#00b3b3',
  padding: 10,
  margin: 15,
  height: 40,
  borderRadius: 25,
  alignItems: 'center',
  justifyContent: 'center',
  flexDirection: 'center'
},
convertButtonText:{
  color: 'white',
  textAlign: 'center'
},
containerTaskBar: {
  flex: 1,
  flexDirection: 'row',
  position: 'absolute',
  justifyContent: 'space-around',
  alignItems: 'center',
  height: '12%',
  width: '100%',
  bottom: 0,
  left: 0,
  borderWidth: 1,
  borderTopColor: 'lightgray',
  backgroundColor: 'white'
},
textTaskBar: {
 width: 188,
 height: 100,
 textAlign: 'center',
 borderWidth: 1,
 padding: 37,
 fontSize: 20,
 borderColor: 'black',
 backgroundColor: 'white'
},
textTaskBarTrack: {
 width: 188,
 height: 100,
 textAlign: 'center',
 borderWidth: 1,
 padding: 37,
 fontSize: 20,
 borderColor: 'black',
 backgroundColor: 'white',
 fontWeight: 'bold',
 color: 'blue'
},
timeInputContainer: {
  flexDirection: 'row',
  justifyContent: 'space-between',
  alignItems: 'center',
  margin: '10%',
  height: 50
}
})
