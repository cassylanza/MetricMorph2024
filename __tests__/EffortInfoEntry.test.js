import React from 'react';
import { render, screen, fireEvent, getByText, getByPlaceholderText, waitFor, changeText, findByPlaceholderText } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { NavigationContainer } from '@react-navigation/native';
import { Actions } from 'react-native-router-flux'; // Mocking react-native-router-flux
import '@testing-library/jest-dom';
import EffortInfoEntry from '../EffortInfoEntry';
import SwimMenu1 from '../SwimMenu1'; // Importing SwimMenu1 component
import TrackMenu1 from '../TrackMenu1'; // Importing TrackMenu1 component
import EffortResults from '../EffortResults';




// Mocking react-native-router-flux actions
jest.mock('react-native-router-flux', () => ({
  Actions: {
    swimmenu1: jest.fn(),
    trackmenu1: jest.fn(),
    convperfinfoentry: jest.fn(),
    conversionresults: jest.fn(),
    splitsinfoentry: jest.fn(),
    splitsresults: jest.fn(),
    effortinfoentry: jest.fn(),
    effortresults: jest.fn(),
  },
}));

describe('EffortInfoEntry', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<EffortInfoEntry />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('task bar swim button appears', () => {
    const { getByText } = render(<EffortInfoEntry />);
    expect(getByText("Swim Menu")).toBeTruthy();
  });

  it('task bar track button appears', () => {
    const { getByText } = render(<EffortInfoEntry />);
    expect(getByText("Track Menu")).toBeTruthy();
  });

  it('calculation button appears', () => {
    const { getByText } = render(<EffortInfoEntry />);
    expect(getByText("CALCULATE")).toBeTruthy();
  })


  describe('EffortInfoEntry component', () => {
    it('displays conversion results when convert button is pressed', async () => {
      const { getByPlaceholderText, queryByText, getByText } = render(<EffortInfoEntry />);

      const currentRaceDistanceInput = getByPlaceholderText(' Current Race Distance (in meters)');
      fireEvent.changeText(currentRaceDistanceInput, '3200');

      const hours = getByPlaceholderText('  hh  ');
      fireEvent.changeText(hours, '00');

      const minutes = getByPlaceholderText('  mm  ');
      fireEvent.changeText(minutes, '06');

      const seconds = getByPlaceholderText('  ss  ');
      fireEvent.changeText(seconds, '05');

      const projectedEffortInput = getByPlaceholderText(' Effort percentage (1-120%)');
      fireEvent.changeText(projectedEffortInput, '80');

      const convertButton = getByText("CALCULATE");
      fireEvent.press(convertButton);

      await waitFor(() => {
        expect(queryByText('65%')).toBeTruthy();
      });
    });
  });



  describe('EffortInfoEntry component', () => {
  it('displays text input boxes for race distance, race time, and effort percentage', () => {
    render(<EffortInfoEntry />);

    // Assert that text input boxes for race distance, race time, and effort percentage are present
    const raceDistanceInput = screen.getByPlaceholderText(' Current Race Distance (in meters)');
    const hours = screen.getByPlaceholderText('  hh  ');
    const minutes = screen.getByPlaceholderText('  mm  ');
    const seconds = screen.getByPlaceholderText('  ss  ');
    const effortPercentageInput = screen.getByPlaceholderText(' Effort percentage (1-120%)');

    expect(raceDistanceInput).toBeTruthy();
    expect(hours).toBeTruthy();
    expect(minutes).toBeTruthy();
    expect(seconds).toBeTruthy();
    expect(effortPercentageInput).toBeTruthy();
  });
});

  test('show swim menu when swim menu button pressed', () => {
    render(
      <NavigationContainer>
        <EffortInfoEntry />
      </NavigationContainer>
    )

    fireEvent.press(screen.getByText('Swim Menu'));

    expect(screen.getByText('Swim Menu')).toBeTruthy();
  });

  test('show track menu when track menu button pressed', () => {
    render(
      <NavigationContainer>
        <EffortInfoEntry />
      </NavigationContainer>
    )

    fireEvent.press(screen.getByText('Track Menu'));

    expect(screen.getByText('Track Menu')).toBeTruthy();
  });




})
