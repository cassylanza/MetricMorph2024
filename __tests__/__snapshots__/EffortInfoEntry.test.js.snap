// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EffortInfoEntry renders correctly 1`] = `
<View>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "marginBottom": "100%",
            "paddingTop": 23,
          }
        }
      >
        <TextInput
          keyboardType="number-pad"
          maxLength={15}
          onChangeText={[Function]}
          placeholder=" Current Race Distance (in meters)"
          placeholderTextColor="grey"
          style={
            {
              "borderBottomColor": "grey",
              "borderBottomWidth": 1,
              "height": 50,
              "margin": "10%",
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
              "margin": "10%",
            }
          }
        >
          <TextInput
            keyboardType="number-pad"
            maxLength={2}
            onChangeText={[Function]}
            placeholder="  hh  "
            placeholderTextColor="grey"
            style={
              {
                "borderBottomColor": "grey",
                "borderBottomWidth": 1,
                "height": 50,
                "margin": "10%",
              }
            }
          />
          <Text>
             : 
          </Text>
          <TextInput
            keyboardType="number-pad"
            maxLength={2}
            onChangeText={[Function]}
            placeholder="  mm  "
            placeholderTextColor="grey"
            style={
              {
                "borderBottomColor": "grey",
                "borderBottomWidth": 1,
                "height": 50,
                "margin": "10%",
              }
            }
          />
          <Text>
             : 
          </Text>
          <TextInput
            keyboardType="decimal-pad"
            maxLength={5}
            onChangeText={[Function]}
            placeholder="  ss  "
            placeholderTextColor="grey"
            style={
              {
                "borderBottomColor": "grey",
                "borderBottomWidth": 1,
                "height": 50,
                "margin": "10%",
              }
            }
          />
        </View>
        <TextInput
          keyboardType="number-pad"
          maxLength={15}
          onChangeText={[Function]}
          placeholder=" Effort percentage (1-120%)"
          placeholderTextColor="grey"
          style={
            {
              "borderBottomColor": "grey",
              "borderBottomWidth": 1,
              "height": 50,
              "margin": "10%",
            }
          }
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "green",
              "borderRadius": 25,
              "flexDirection": "center",
              "height": 40,
              "justifyContent": "center",
              "margin": 15,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "textAlign": "center",
              }
            }
          >
             CALCULATE 
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "alignItems": "center",
        "bottom": 0,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "position": "absolute",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "backgroundColor": "white",
            "borderColor": "black",
            "borderWidth": 1,
            "fontSize": 20,
            "height": 100,
            "padding": 37,
            "textAlign": "center",
            "width": 188,
          }
        }
      >
         Swim Menu 
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "backgroundColor": "white",
            "borderColor": "black",
            "borderWidth": 1,
            "color": "blue",
            "fontSize": 20,
            "fontWeight": "bold",
            "height": 100,
            "padding": 37,
            "textAlign": "center",
            "width": 188,
          }
        }
      >
         Track Menu 
      </Text>
    </View>
  </View>
</View>
`;
